src/rgw/rgw_http_errors.h

```diff
@@ -10,69 +10,69 @@ struct rgw_http_errors {
+    { ERR_QUOTA_EXCEEDED, 403, "QuotaExceeded", "Quota Exceeded" }, 
+    { ENOENT, 404, "NoSuchKey", "The resource you requested does not exist" },
+    { ERR_NO_SUCH_BUCKET, 404, "NoSuchBucket", "The specified bucket does not exist." },
+    { ERR_NO_SUCH_WEBSITE_CONFIGURATION, 404, "NoSuchWebsiteConfiguration", "No such website configuration" },
+    { ERR_NO_SUCH_UPLOAD, 404, "NoSuchUpload", "The specified multipart upload does not exist. The upload ID might be invalid, or the multipart upload might have been aborted or completed." },
+    { ERR_NOT_FOUND, 404, "Not Found", "Not Found"},
+    { ERR_NO_SUCH_LC, 404, "NoSuchLifecycleConfiguration", "The lifecycle configuration does not exist." },
+    { ERR_NO_ROLE_FOUND, 404, "NoSuchEntity", "The role does not exist." },
+    { ERR_METHOD_NOT_ALLOWED, 405, "MethodNotAllowed", "The specified method is not allowed against this resource." },
+    { ETIMEDOUT, 408, "RequestTimeout", "Your socket connection to the server was not read from or written to within the timeout period." },
+    { EEXIST, 409, "BucketAlreadyExists", "The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again." },
+    { ERR_USER_EXIST, 409, "UserAlreadyExists", "The requested user name is not available. Please select a different name and try again." }, // doesn't exist in s3 error codes
+    { ERR_EMAIL_EXIST, 409, "EmailExists", "The request email is not available. Please select a different email and try again." }, // doesn't exist in s3 error codes
+    { ERR_KEY_EXIST, 409, "KeyExists", "The request key is not available. Please select a different key and try again." },  // doesn't exist in s3 error codes
+    { ERR_ROLE_EXISTS, 409, "EntityAlreadyExists", ""},
+    { ERR_DELETE_CONFLICT, 409, "DeleteConflict", ""},
+    { ERR_INVALID_SECRET_KEY, 400, "InvalidSecretKey", "The provided security credentials are not valid."},
+    { ERR_INVALID_KEY_TYPE, 400, "InvalidKeyType", "Invalid Key Type"},
+    { ERR_INVALID_CAP, 400, "InvalidCapability", "Invalid Capability"},
+    { ERR_INVALID_TENANT_NAME, 400, "InvalidTenantName", "Invalid Tenant Name" },
+    { ENOTEMPTY, 409, "BucketNotEmpty", "The bucket you tried to delete is not empty." },
+    { ERR_PRECONDITION_FAILED, 412, "PreconditionFailed", "At least one of the preconditions you specified did not hold." },
+    { ERANGE, 416, "InvalidRange", "The requested range cannot be satisfied." },
+    { ERR_UNPROCESSABLE_ENTITY, 422, "UnprocessableEntity", "Unprocessable Entity" },
+    { ERR_LOCKED, 423, "Locked", "Locked" },
+    { ERR_INTERNAL_ERROR, 500, "InternalError", "We encountered an internal error. Please try again." },
+    { ERR_NOT_IMPLEMENTED, 501, "NotImplemented", "A header you provided implies functionality that is not implemented." },
+    { ERR_SERVICE_UNAVAILABLE, 503, "ServiceUnavailable", "Reduce your request rate."}
```

Comment:
@Liuchang0812 "ServiceUnavailable", "Reduce your request rate." Is this correct description  of error code?

Reply:
It sounds like if we are aiming for completeness we'd have to add a new error code `{ ERR_SLOW_DOWN, 503, "SlowDown", "Reduce your request rate."}` and change the text of the current one to generic `"Service unavailable."` text. It looks like currently radosgw never returns a 503 code for any reason, so this would just be for the sake of having the error constants defined in case it comes up in the future.